core/tools/coverage-report

```diff
@@ -1,25 +1,72 @@
 #!/bin/sh
 
-COVERAGE_THRESHOLD=${COVERAGE_THRESHOLD:-0}
+# Configurable variables
+readonly COVERAGE_THRESHOLD=${COVERAGE_THRESHOLD:-0}
+readonly WHOLE_SRC_COVERAGE=${WHOLE_SRC_COVERAGE:-1}
+readonly MOVE_FROM_SRC=${MOVE_FROM_SRC:-1}
 
-coverage run --source=./src /dev/null 2>/dev/null
-mv .coverage .coverage.empty
-coverage combine .coverage.*
+if [ ${COVERAGE_THRESHOLD} -ne 0 ]; then
+  echo "COVERAGE_THRESHOLD set to ${COVERAGE_THRESHOLD}%"
+fi
+
+# Optionally move coverage files from src
+# (tests usually save it there).
+if [ ${MOVE_FROM_SRC} -eq 1 ]; then
```

Comment:
why wouldn't we do this? isn't the "if ls" check sufficient?

Reply:
> are you suggesting completely removing the if statement?

yes, i'm not a fan of configuration options covering scenarios that we don't ever encounter :)

in the tests.yml case, you shouldn't have any coverages in the `core` subdir, so the "if ls" check covers you there; strictly speaking, you could just run the `mv src/.coverage.* . 2>/dev/null` and ignore any errors, right?

Reply:
Actually, in tests, we _have_ all the coverage files in `core` already, as each test moves it there from `src` in the artifacts - so nothing would be moved in that case.

Just `mv src/.coverage.* . 2>/dev/null` is certainly doable, but then we lose `echo` confirming the move was done, and more importantly the output of `ls` telling us what files were moved. Probably not vital, but if we can give this information, why hide it.

Reply:
yes, sorry, i meant `core/src`, because the local checkout didn't run any tests so everything we get must be coming from the artifacts

> then we lose echo confirming the move was done

`mv -v`? :)

(i don't have a problem with the "if ls" check per se, but now it turns out that `mv -v ...` would actually do mostly the same thing you want, so :woman_shrugging: )

Reply:
Deleting the variable and usage of `ls` in [f246a1f](https://github.com/trezor/trezor-firmware/pull/2857/commits/f246a1f5f5a9f1ac1251e9026add3358696b9bef)